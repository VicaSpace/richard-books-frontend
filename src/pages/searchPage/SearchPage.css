.search-page-container {
  padding-top: 1.1875rem;
}

.search-page-slider-heading {
  padding-left: 1.5rem;
  padding-bottom: 1.1875rem;
  font-size: 1rem;
}
