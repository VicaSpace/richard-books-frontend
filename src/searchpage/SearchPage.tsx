import { Container } from '@chakra-ui/react'
import React from 'react'

const SearchPage: React.FC<{}> = () => {
  return (
    <Container>
      <p>SearchPage</p>
    </Container>
  )
}

export default SearchPage
