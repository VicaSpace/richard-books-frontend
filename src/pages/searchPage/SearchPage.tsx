import React from 'react'
import './SearchPage.css'

const SearchPage: React.FC<{}> = () => {
  return <>Search Page</>
}

export default SearchPage
