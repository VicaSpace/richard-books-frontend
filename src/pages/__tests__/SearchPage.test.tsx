import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter, Route, Routes } from 'react-router'
import SearchPage from '../searchPage/SearchPage'

const books = [
  {
    id: 1,
    name: 'Book 1',
    img: 'img src',
    genre: 'Children',
    description: 'Description',
    price: 5,
    Author: {
      id: 1,
      name: 'John',
      img: "John's image"
    }
  },
  {
    id: 2,
    name: 'Book 2',
    img: 'img src',
    genre: 'Children',
    description: 'Description',
    price: 5,
    Author: {
      id: 1,
      name: 'John',
      img: "John's image"
    }
  }
]

describe('test search page', () => {
  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/search']}>
        <Routes>
          <Route path="/search" element={<SearchPage books={books} />} />
        </Routes>
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
