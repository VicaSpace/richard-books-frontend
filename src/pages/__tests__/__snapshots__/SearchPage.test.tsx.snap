// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test search page should render correctly 1`] = `
<div>
  <div
    class="search-page-container"
  >
    <div
      class="search-section-container"
    >
      <form
        class="search-section-form"
      >
        <svg
          class="search-section-icon"
          fill="currentColor"
          height="0.66rem"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="0.66rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M18.031 16.617l4.283 4.282-1.415 1.415-4.282-4.283A8.96 8.96 0 0 1 11 20c-4.968 0-9-4.032-9-9s4.032-9 9-9 9 4.032 9 9a8.96 8.96 0 0 1-1.969 5.617zm-2.006-.742A6.977 6.977 0 0 0 18 11c0-3.868-3.133-7-7-7-3.868 0-7 3.132-7 7 0 3.867 3.132 7 7 7a6.977 6.977 0 0 0 4.875-1.975l.15-.15z"
            />
          </g>
        </svg>
        <input
          placeholder="Search"
          value=""
        />
      </form>
      <div
        class="search-section-recent-search-container"
      >
        <div
          class="search-section-recent-search-heading"
        >
          Recent Searches
        </div>
        <div
          class="search-section-recent-search"
        />
      </div>
    </div>
    <div
      class="search-page-slider-container"
    >
      <div
        class="search-page-slider-heading"
      >
        Search Results
      </div>
      <div
        class="book-slider-container"
      />
    </div>
    <div
      class="search-page-slider-container"
    >
      <div
        class="search-page-slider-heading"
      >
        All Books
      </div>
      <div
        class="book-slider-container"
      >
        <div
          class="book-slider-item"
        >
          <div
            class="book-item-container"
          >
            <div
              class="book-item-rating-container"
            >
              <div
                class="book-item-rating"
              >
                <svg
                  class="book-item-star"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
                4.5
              </div>
            </div>
            <img
              alt="Book 1"
              class="book-item-img"
              src="img src"
            />
            <div
              class="book-item-name"
            >
              Book 1
            </div>
            <div
              class="book-item-author"
            >
              John
            </div>
            <div
              class="book-item-price"
            >
              <span
                class="book-item-emphasize"
              >
                $
                5
              </span>
              .00
            </div>
          </div>
        </div>
        <div
          class="book-slider-item"
        >
          <div
            class="book-item-container"
          >
            <div
              class="book-item-rating-container"
            >
              <div
                class="book-item-rating"
              >
                <svg
                  class="book-item-star"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
                4.5
              </div>
            </div>
            <img
              alt="Book 2"
              class="book-item-img"
              src="img src"
            />
            <div
              class="book-item-name"
            >
              Book 2
            </div>
            <div
              class="book-item-author"
            >
              John
            </div>
            <div
              class="book-item-price"
            >
              <span
                class="book-item-emphasize"
              >
                $
                5
              </span>
              .00
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="navbar-container"
    >
      <img
        class="navbar-icon"
        src="home-dimmed.svg"
      />
      <img
        class="navbar-icon"
        src="search.png"
      />
      <img
        class="navbar-icon"
        src="book-dimmed.svg"
      />
    </div>
  </div>
</div>
`;
